spec/datadog/tracing/transport/traces_spec.rb

```diff
@@ -283,59 +283,113 @@
     end
 
     context 'for native span event support by the agent' do
-      context 'on a successful agent info call' do
-        context 'with support not advertised' do
-          let(:native_events_supported) { nil }
-
-          it 'does not encode native span events' do
-            expect(Datadog::Tracing::Transport::Traces::Chunker).to receive(:new).with(
-              encoder_v2,
-              native_events_supported: false
-            ).and_return(chunker)
-            send_traces
+      after do
+        Datadog.configuration.tracing.reset_options!
+      end
+      context 'when native_span_events option is configured' do
+        context 'when set to true' do
+          before do
+            Datadog.configuration.tracing.native_span_events = true
           end
-        end
 
-        context 'with support advertised as supported' do
           let(:native_events_supported) { true }
 
-          it 'encodes native span events' do
+          it 'uses the configured value' do
             expect(Datadog::Tracing::Transport::Traces::Chunker).to receive(:new).with(
               encoder_v2,
               native_events_supported: true
             ).and_return(chunker)
+
+            send_traces
+          end
+
+          it 'does not query the agent' do
+            allow(Datadog::Tracing::Transport::Traces::Chunker).to receive(:new).and_return(chunker)
+            expect_any_instance_of(Datadog::Core::Environment::AgentInfo).not_to receive(:fetch)
+
             send_traces
           end
```

Comment:
:+1: Nice, protects against accidentally still calling the agent

Reply:
👍